File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 8/14/2025, 3:50:03 PM
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        m=0
        for i in range(len(s)):
            a=""
            for j in range(i,len(s)):
                if s[j] not in a:
                    a+=s[j]
                else:
                    a=""
                    break
                if len(a)>m:
                    m=len(a)
        return m

        

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 8/14/2025, 3:49:58 PM
class Solution {
    public int lengthOfLongestSubstring(String s) {
       
        int freq[]=new int[255];
        int i=0;
        int j=0;
        int ans=0;
        while(j<s.length()){
            freq[s.charAt(j)]++;
            while(freq[s.charAt(j)]>1){
                freq[s.charAt(i)]--;
                i++;
            }
            ans=Math.max(ans,j-i+1);
            j++;
        }
        return ans;
        
    }
}